microprofile/WFLY-12609_MicroProfile_JWT_1_1.adoc

```diff
@@ -0,0 +1,109 @@
+= WFLY-12609 MicroProfile JWT 1.1 Support
+:author:            Darran Lofthouse
+:email:             darran.lofthouse@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This proposal is in relation to adding support to WildFly for the MicroProfile JWT 1.1 specification.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-12609[WFLY-12609]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1346[EAP7-1346]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+The primary integration will be happening by the addition of a new subsystem `microprofile=jwt` within https://github.com/wildfly/wildfly[WildFly].
+
+Additional integration may be added within https://github.com/wildfly-security/wildfly-elytron[WildFly Elytron] to bridge into the application server's security framework.
+
+=== Other Interested Projects
+
+The integration will also be making use of https://github.com/smallrye/smallrye-jwt[SmallRye JWT] as well as pulling in the https://github.com/eclipse/microprofile-jwt-auth[Eclipse MicroProfile JWT] API library.
+
+== Requirements
+
+=== Hard Requirements
+
+Primarily this enhancement will be delivered using a new subsystem `microprofile-jwt` which will be added to WildFly.  The subsystem will install a DeploymentUnitProcessor which will detect if MicroProfile JWT is required and activate a suitable security configuration as described within this proposal.
```

Comment:
1. Will this subsystem have any operations available in management interface? If it will - which?
1. Will this subsystem have any attributes interface? If it will - which?

Reply:
No the subsystem will not have any operations or attributes at this point in time, the presence of the subsystem will activate it's DUPs and the remainder of the configuration will be based on MP Config resolution rules.

In the future we could add some configuration for alternative Elytron integration options but those should not be required at the outset.